src/Psalm/Internal/Provider/ReturnTypeProvider/ArrayColumnReturnTypeProvider.php

```diff
@@ -114,6 +114,8 @@ public static function getFunctionReturnType(FunctionReturnTypeProviderEvent $ev
                 }
                 //array_column skips undefined elements so resulting type is necessarily defined
                 $result_element_type->possibly_undefined = false;
+            } elseif ($value_column_name === null) {
+                $result_element_type = $row_shape;
```

Comment:
$result_element_type should be a Union, $row_shape is an Atomic (a part of a Union). `new Union([$row_shape]);` should be ok

Reply:
So my reasoning was right after all 😄 
Thank you I fixed that 💪

Reply:
I don't understand why? `$result_element_type ` should be `$row_shape`. How is wrapping it in a single element union useful?

Reply:
@alex-dev This is a minor weirdness in the way Psalm handles types, right now everything is a `Union` which has one or more `Atomic`s. I've been slowly working on changing it so that types can be `Atomic`s or `Union`s or `Intersection`s, but it's a long endeavor.

Reply:
I see. Thank for the explanation.